tensorboard/defs/defs.bzl

```diff
@@ -98,50 +100,94 @@ def tf_ts_library(strict_checks = True, **kwargs):
 
     if strict_checks == False:
         tsconfig = "//:tsconfig-lax"
-    elif "test_only" in kwargs and kwargs.get("test_only"):
-        tsconfig = "//:tsconfig-test"
     kwargs.setdefault("deps", []).extend(["@npm//tslib", "//tensorboard/defs:strict_types"])
 
-    ts_library(tsconfig = tsconfig, supports_workers = True, **kwargs)
+    new_srcs = []
+    # Find test.ts files and rename to test.spec.ts to be compatible with
+    # spec_bundle() tooling.
+    for s in srcs:
+      if s.endswith("_test.ts") or s.endswith("-test.ts"):
+        # Make a copy of the file with name .spec.ts and switch to that as
+        # the src file.
+        new_src = s[0:s.rindex('.ts')] + ".spec.ts"
+        native.genrule(
+            name = new_src + '_spec_copy',
+            srcs = [s],
+            outs = [new_src],
+            cmd = "cat $(SRCS) > $@",
+        )
+        new_srcs.append(new_src)
+      else:
+        # Not a test file. Nothing to do here.
+        new_srcs.append(s)
 
-def tf_ng_web_test_suite(runtime_deps = [], bootstrap = [], deps = [], **kwargs):
+    ts_library(
+        srcs = new_srcs,
+        tsconfig = tsconfig,
+        supports_workers = True,
+        # Override prodmode_target, devmode_target, and devmode_module to be
+        # compatible with extract_js_module_output() and spec_bundle() tooling.
+        # See the angular/components example:
+        # https://github.com/angular/components/blob/871f8f231a7a86a7a0778e345f4d517109c9a357/tools/defaults.bzl#L114-L121
+        prodmode_target = "es2020",
+        devmode_target = "es2020",
+        devmode_module = "esnext",
+        **kwargs)
+
+def tf_ng_web_test_suite(name, deps = [], **kwargs):
```

Comment:
I am confused here. Why do all of our calls to tf_ng_web_test_suite already have a name specified when the old definition did not have name as a parameter?

Reply:
The `name` argument was previously captured by the `**kwargs` argument and automatically passed down to karma_web_test_suite unmodified. Now that we are using `name` argument explicitly in tf_ng_web_test_suite then we have to explicitly list it in the arguments here.

Reply:
Ah that makes sense. Thanks.